### PR TITLE
Add support for deserialization of types that cannot be constructed directly via a cast from string

### PR DESCRIPTION
**Pull request by janhohenheim**

### CODE REVIEW

File: LinqToRest.Test/QueryProviderTest.cs [hunk 1]
```diff
@@ -20,7 +20,7 @@ public void ReturnsRestQuery()
 
             Assert.Equal(expected, actual);
         }
-
+
```
Comment (dngroth) [inline LinqToRest.Test/QueryProviderTest.cs]: Remove whitespaces

File: LinqToRest.Test/QueryProviderTest.cs [hunk 2]
```diff
@@ -139,6 +151,33 @@ private static HttpClient MockHttpClient()
                 new EntityWithQueryableMember("Test2", CreateQuery<EntityWithSimpleMembers>("entitywithqueryablemembers/Test2/")),
             });
 
+
```
Comment (dngroth) [inline LinqToRest.Test/QueryProviderTest.cs:153]: Newlines!

File: changelog.md [hunk 3]
```diff
@@ -16,3 +16,7 @@
 - Rename `QueryProviderBuilder` to `QueryableBuilder`
 - Let `QueryableBuilder` return an `IQueryable<T>` directly
 - Provide an interface for `QueryableBuilder`
+
+## TBA
+
```
Comment (dngroth) [inline changelog.md]: Remove newline

**Review by dngroth — CHANGES_REQUESTED** [review]

File: changelog.md (hunk 3 above)
Comment (janhohenheim) [reply changelog.md]: Addressed in https://github.com/messerli-informatik-ag/linq-to-rest/pull/27/commits/16f9f717fa3e4ec123a5107ee8fc422fa04e6ff8

File: LinqToRest.Test/QueryProviderTest.cs (hunk 1 above)
Comment (janhohenheim) [reply LinqToRest.Test/QueryProviderTest.cs]: Addressed in https://github.com/messerli-informatik-ag/linq-to-rest/pull/27/commits/c6de22bdf8e765326f25d3e6fc03ce169e79b74a

File: LinqToRest.Test/QueryProviderTest.cs (hunk 2 above)
Comment (janhohenheim) [reply LinqToRest.Test/QueryProviderTest.cs:153]: Addressed in https://github.com/messerli-informatik-ag/linq-to-rest/pull/27/commits/d2ca814d94c0c701fd8dc5dd7d9e32e4274521ae

**Review by dngroth — APPROVED** [review]

**Pull request merged**